brainzutils/metrics.py

```diff
@@ -29,172 +29,51 @@ def decorated(*args, **kwargs):
 
 @cache.init_required
 @metrics_init_required
-def increment(metric_name, amount=1):
-    """Increment the counter for a metric by a set amount. A metric is a counter that can increment over time
-    and can be used for monitoring any statistic.
-
-    If incrementing the counter causes it to go over redis' internal counter limit of 2**63-1, the counter
-    is reset to 0. The metric name ``tag`` is reserved and cannot be used.
-
-    Arguments:
-        metric_name: the name of a metric
-        amount: the amount to increase the counter by, must be 1 or greater (default: 1)
-
-    Raises:
-        ValueError if amount is less than 1 or greater than 2**63-1
-        ValueError if the reserved metric name ``tag`` is used
+def set(metric_name: str, tags: Dict[str,str]={}, timestamp: int=None, **fields):
+    """
+        Submit a metric to the MetaBrainz influx datastore for graphing/monitoring
+        purposes.
+
+        Args:
+          metric_name: The name of the metric to record.
+          tags: Additional influx tags to write with the metric. (optional)
+          timestamp: A nanosecond timestamp to use for this metric. If not provided
+                     the current time is used.
+          fields: The key, value pairs to store with this metric.
     """
 
-    if amount < 1:
-        raise ValueError("amount must be 1 or greater")
-    if amount > REDIS_MAX_INTEGER:
-        raise ValueError("amount is too large")
-    if metric_name in RESERVED_TAG_NAMES:
-        raise ValueError("the name '{}' is reserved".format(metric_name))
-
+    # Add types to influx data
     try:
-        ret = cache.hincrby(_metrics_project_name, metric_name, amount, namespace=NAMESPACE_METRICS)
-    except ResponseError as e:
-        # If the current value is too large, redis will return this error message.
-        # Reset to 0 and re-increment
-        if e.args and "increment or decrement would overflow" in e.args[0]:
-            cache.hset(_metrics_project_name, metric_name, 0, namespace=NAMESPACE_METRICS)
-            ret = cache.hincrby(_metrics_project_name, metric_name, amount, namespace=NAMESPACE_METRICS)
+        host = os.environ['PRIVATE_IP']
+    except KeyError:
+        host = socket.gethostname()
+
+    tags["dc"] = "hetzner"
+    tags["server"] = host
+    tags["project"] = _metrics_project_name
+    tag_string = ",".join([ "%s=%s" % (k, tags[k]) for k in tags ])
+
+    fields_list = []
+    for k in fields:
```

Comment:
Nit: can do like `for key, value in fields.items()` to avoid repeating `fields[k]` in following lines.